owners/src/owners_tree.js

```diff
@@ -87,6 +87,31 @@ class OwnersTree {
     return this.rules.concat(parentRules);
   }
 
+  /**
+   * Provides a list of ownership rules for a file.
+   *
+   * Includes the node's own rules as well as inherited rules, in decreasing
+   * order of depth (ie. the root OWNERS rules will be last).
+   *
+   * @param {!string} filename file to get rules for.
+   * @return {OwnersRule[]} list of all ownership rules for the file.
+   */
+  fileRules(filename) {
+    return this.allRules.filter(rule => rule.matchesFile(filename));
+  }
+
+  /**
+   * Provides a list of owners rules for a file.
+   *
+   * @param {!string} filename file to get owners for.
+   * @return {Owner[]} list of all owners for the file.
+   */
+  fileOwners(filename) {
+    return this.fileRules(filename)
+      .map(rule => rule.owners)
+      .reduce((left, right) => left.concat(right), []);
```

Comment:
Do you need this? `Array.prototype.map()` already returns an array.

Reply:
`rule.owners` is an array, this flattens the array of arrays.

Reply:
Ah okay. No need to check for dupes?

Reply:
Not here, dupes aren't relevant because even if identical rules exist they're at different precedence, so order/depth is important